### PR TITLE
VN-71 State detection and install button

### PR DESCRIPTION
**Pull request by drawm**
### Features
* Maintenance app can call bash scripts to:
  - Detect application state
  - Setup/build docker images
* Install button is only enabled when the `veto` docker image is not present on the system
* A throttle mechanism prevent the user from spamming the install button

### Fix
* Remove postinstall script because they crash on new installation (ie docker build)
* Fix state detection 
  - Wrong format for "unknown" state
  - Missing "created" state

### Test
Tested manually
#### Steps
* `yarn start:maintenance`
* Navigate to `localhost:8181`
* Login to veto (psw `veto`)
* Veto state should tell you if the veto image is created or not (unknown if no information is available)
* If a `veto` image is not available, the `install` button should be available
* When the `install` button is clicked, the backend should start building the veto docker image (this can take a few minutes)
* When it is done, the `install` button should be disabled veto's state should be `created`
* If you start a veto instanced names `veto` the state should change to `running`

#### Other behaviors
* If you remove the veto container (`docker image rm veto`), its state should automatically change to `unknown` 
* If you stop the veto container (`docker stop veto`), its state should automatically change to `stopped`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Vagrant
```
Comment (drawm) [inline .gitignore:1]: Leftover, will come in handy soon-ish

File: package.json [hunk 2]
```diff
@@ -36,7 +38,7 @@
     "build:shared": "yarn workspace @veto/config build",
     "test:integration": "jest --testPathPattern='./packages/.+/tests/integration/'",
     "pretest:unit": "yarn build:shared",
-    "test:unit": "jest --testPathIgnorePatterns='packages/.+/tests/integration/'"
+    "test:unit": "jest --testPathIgnorePatterns='packages/.+(/out|/tests/integration/)'"
```
Comment (drawm) [inline package.json:41]: Prevent looking for tests in `/out` folders

File: packages/maintenance-app/backend/src/index.ts [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
 // TODO: Move to ./utils
-function isInstalled(applicationState: ApplicationState): boolean {
-  if ([State.UNKNOWN, State.MISSING].includes(applicationState.cyphernode)) {
-    return false
-  }
-
-  if ([State.UNKNOWN, State.MISSING].includes(applicationState.veto)) {
-    return false
-  }
+function isInstalled(state: State): boolean {
```
Comment (drawm) [inline packages/maintenance-app/backend/src/index.ts:13]: Will be moved to a helper file when we start adding more routes

File: packages/maintenance-app/backend/src/scripts/utils/run.ts [hunk 4]
```diff
@@ -2,19 +2,31 @@ import * as path from 'path'
...
 
-export default function run(script: string): Promise<string> {
+export type EnvironmentVariables = {
+  [name: string]: boolean | string | number
+}
+
+export default function run(script: string, env?: EnvironmentVariables): Promise<string> {
+  // need to set the current path properly before starting a script
   const pathToScript: string = path.normalize(config.scriptLocation + script)
+  const options = env ? { env: { PATH: process.env.PATH, ...env } } : null
```
Comment (drawm) [inline packages/maintenance-app/backend/src/scripts/utils/run.ts:12]: `PATH` is needed to call globally accessible scripts on the host machine (ie docker)

File: packages/maintenance-app/frontend/src/pages/Dashboard/index.tsx [hunk 5]
```diff
@@ -1,13 +1,119 @@
...
+  applicationState: ApplicationState
+}
+
+type ApplicationControlProps = {
+  start?: () => void
+  install: () => void
+  factoryReset?: () => void
+  applicationState: ApplicationState
+}
+const ApplicationControls = (props: ApplicationControlProps) => {
```
Comment (drawm) [inline packages/maintenance-app/frontend/src/pages/Dashboard/index.tsx:61]: Will be moved to its own file when we start adding more buttons/controls

File: packages/shared/config/package.json [hunk 6]
```diff
@@ -8,8 +8,8 @@
   "main": "out/index.js",
   "type": "out/index.d.ts",
   "scripts": {
-    "postinstall": "tsc -p . --outDir out",
     "start": "tsc -p . --outDir out --watch",
+    "prebuild": "rm -fr out/*",
```
Comment (drawm) [inline packages/shared/config/package.json:12]: Needed to force retranspilation (without it, type changes are not compiled if no code has been changed)

File: scripts/setup.sh [hunk 7]
```diff
@@ -7,6 +7,7 @@ VERSION_NUMBER=${VERSION_NUMBER:-"latest"}
 CURRENT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 CYPHERNODE_DIR=${CYPHERNODE_DIR:-./cyphernode}
 ORIGINAL_DIR=$(pwd)
+DOCKER_SOURCE_DIR=${DOCKER_SOURCE_DIR:-$CURRENT_DIR }
```
Comment (drawm) [inline scripts/setup.sh:10]: All those variables are getting out of hand, any suggestion on better names?

Comment (nk1tz) [reply scripts/setup.sh:10]: Doesn't look too bad from my perspective. :)

Comment (nk1tz) [reply scripts/setup.sh:10]: But glad your thinking about it :+1:

Comment (drawm) [reply scripts/setup.sh:10]: `ORIGINAL_DIR` & `CURRENT_DIR` could be better named IMO

`CURRENT_DIR` = Folder where the file is
`ORIGINAL_DIR` = I think its the current working directory of the process who called the script (not 100% sure, I need to double check)

Comment (drawm) [reply scripts/setup.sh:10]: Any idea on better name?

**Pull request merged**